File: EduQuiz.Web/starter-vite-ts/src/types/flashcard.ts
import type { Visibility } from "./quiz";

export type FlashcardDeck = {
  id?: string;
  title: string;
  visibility: Visibility;
  flashcards: Flashcard[];
}

export type Flashcard = {
  id?: string;
  frontSideText: string;
  backSideText: string;
  hint?: string;
}

export type AIConfigParams = {
  subject: string;
  topic: string;
  language: string;
  numFlashcards: string;
}

File: EduQuiz.AI/src/api/endpoints/flashcard.py
from fastapi import APIRouter, HTTPException
from fastapi.responses import StreamingResponse
from src.api.models.flashcard import FlashcardRequest
from src.services.ai_service import stream_flashcard

router = APIRouter()

@router.post("/generate")
async def generate_flashcard_endpoint(request: FlashcardRequest):
    try:
        return StreamingResponse(
            stream_flashcard(
                subject=request.subject,
                topic=request.topic,
                num_flashcards=request.numFlashcards,
                language=request.language,
            ),
            media_type="application/x-ndjson",
        )

    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Error generating flashcards: {str(e)}")